neuro_san/internals/messages/base_message_dictionary_converter.py

```diff
@@ -0,0 +1,168 @@
+
+# Copyright (C) 2023-2025 Cognizant Digital Business, Evolutionary AI.
+# All Rights Reserved.
+# Issued under the Academic Public License.
+#
+# You can be released from the terms, and requirements of the Academic Public
+# License by purchasing a commercial license.
+# Purchase of a commercial license is mandatory for any use of the
+# neuro-san SDK Software in commercial settings.
+#
+# END COPYRIGHT
+
+from typing import Any
+from typing import Dict
+from typing import List
+
+from langchain_core.messages.ai import AIMessage
+from langchain_core.messages.base import BaseMessage
+from langchain_core.messages.human import HumanMessage
+from langchain_core.messages.system import SystemMessage
+from langchain_core.messages.tool import ToolMessage
+
+from leaf_common.serialization.interface.dictionary_converter import DictionaryConverter
+
+from neuro_san.internals.messages.agent_message import AgentMessage
+from neuro_san.internals.messages.agent_framework_message import AgentFrameworkMessage
+from neuro_san.internals.messages.agent_tool_result_message import AgentToolResultMessage
+from neuro_san.internals.messages.chat_message_type import ChatMessageType
+
+
+class BaseMessageDictionaryConverter(DictionaryConverter):
+    """
+    A DictionaryConverter implementation which can convert langchain BaseMessages
+    back and forth to our own ChatMessage dictionary format (as defined in chat.proto).
+    """
```

Comment:
Adhere to leaf-common DictionaryConverter interface, as that is really what is going on here - back and forth from a particular type.